examples/confluent-server-oauth/README.md

```diff
@@ -0,0 +1,93 @@
+*without** LDAP.
+
+## Getting Started
+
+To run the Docker Compose setup, execute the following command:
+
+```bash
+docker compose up -d
+```
+
+This will:
+
+1. Start a Keycloak server on port 8080. This acts as the OAuth identity provider.
+2. Run Confluent Server with Role-based access control (RBAC).
+3. Create a new listener `EXTERNAL` for SASL/OAuthBearer in Confluent Server.
+4. Configure Confluent Server to connect to local Keycloak identity provider.
+
+## RBAC using OAuth
+
+1. Get the Kafka cluster ID from the `CLUSTER_ID` environment variable defined in the `docker-compose.yml` file, or via the following GET REST request: http://localhost:8091/v1/metadata/id
+
+2. Get an access token from Keycloak passing the client credentials of super user client app. The client credentials are base64 encoded `client_id:client_secret` and is configured in `keycloak-realm-export.json`.
+
+    ```shell
+    curl -X POST \
+       -H "Authorization: Basic c3VwZXJ1c2VyX2NsaWVudF9hcHA6c3VwZXJ1c2VyX2NsaWVudF9hcHBfc2VjcmV0" \
+       -H "Content-Type: application/x-www-form-urlencoded" \
+       -d "grant_type=client_credentials" \
+       http://localhost:8080/realms/cp/protocol/openid-connect/token
+    ```
+
+3. Grant access to other client apps or users by assigning roles to them.
+   For example, granting access to client app `client_app1` over topic `test` of Kafka cluster `vHCgQyIrRHG8Jv27qI2h3Q`, replace `<access-token>` with token obtained above:
+
+    ```shell
+    curl -v \
+      -H "Authorization: Bearer <access-token>" \
+      -H "Content-Type: application/json" \
+      -H "Accept: application/json" \
+      -X POST 'http://localhost:8091/security/1.0/principals/User:client_app1/roles/ResourceOwner/bindings' \
+      -d '{"scope":{"clusters":{"kafka-cluster":"vHCgQyIrRHG8Jv27qI2h3Q"}}, "resourcePatterns":[{"resourceType":"Topic", "name":"test", "patternType":"LITERAL"}]}'
+    ```
+
+   Also assign access to a consumer group.
+   ```shell
+    curl -v \
+      -H "Authorization: Bearer <access-token>" \
+      -H "Content-Type: application/json" \
+      -H "Accept: application/json" \
+      -X POST 'http://localhost:8091/security/1.0/principals/User:client_app1/roles/ResourceOwner/bindings' \
+      -d '{"scope":{"clusters":{"kafka-cluster":"vHCgQyIrRHG8Jv27qI2h3Q"}}, "resourcePatterns":[{"resourceType":"Group", "name":"console-consumer-group", "patternType":"LITERAL"}]}'
+    ```
+
+## Produce and Consume to Kafka using OAuth
+
+1. Create a client configuration file `client.properties` that uses OAuth:
+
+    ```properties
+    sasl.mechanism=OAUTHBEARER
+    security.protocol=SASL_PLAINTEXT
+    group.id=console-consumer-group
+    sasl.login.callback.handler.class=org.apache.kafka.common.security.oauthbearer.secured.OAuthBearerLoginCallbackHandler
+    sasl.oauthbearer.token.endpoint.url=http://localhost:8080/realms/cp/protocol/openid-connect/token
```

Comment:
I tried using this token end point in my producer sample. I am getting a 400 error from the token endpoint as response.

Reply:
Share the curl request you tried.